src/test/java/uk/gov/companieshouse/service/AppealServiceTest.java

```diff
@@ -0,0 +1,114 @@
+package uk.gov.companieshouse.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.companieshouse.model.Appeal;
+import uk.gov.companieshouse.repository.AppealRepository;
+import uk.gov.companieshouse.util.TestUtil;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+public class AppealServiceTest {
+
+    private static final String TEST_RESOURCE_ID = "1";
+    private static final String TEST_ERIC_ID = "1";
+    public static final String TEST_COMPANY_ID = "12345678";
+    public static final String TEST_PENALTY_REFERENCE = "A12345678";
+    public static final String TEST_REASON_TITLE = "This is a title";
+    public static final String TEST_REASON_DESCRIPTION = "This is a description";
+
+    @InjectMocks
+    private AppealService appealService;
+
+    @Mock
+    private AppealRepository appealRepository;
+
+    @Test
+    public void testCreateAppeal_returnsResourceId() throws Exception {
+
+        ArgumentCaptor<Appeal> appealArgumentCaptor = ArgumentCaptor.forClass(Appeal.class);
+
+        Appeal appeal = TestUtil.getValidAppeal();
+        appeal.setId(TEST_RESOURCE_ID);
+
+        when(appealRepository.insert(any(Appeal.class))).thenReturn(appeal);
+
+        appealService.saveAppeal(appeal, TEST_ERIC_ID);
+
+        verify(appealRepository).insert(appealArgumentCaptor.capture());
+
+        assertAll("Create appeal sets created by and created at",
```

Comment:
I wonder what benefit of using `assertAll` you see. I never used it so I am not sure how it helps compared to having assertions one under another.

Reply:
Priority: do not change for now.